exposed-dao/src/main/kotlin/org/jetbrains/exposed/dao/EntityClass.kt

```diff
@@ -77,7 +78,10 @@ abstract class EntityClass<ID : Comparable<ID>, out T: Entity<ID>>(val table: Id
     fun removeFromCache(entity: Entity<ID>) {
         val cache = warmCache()
         cache.remove(table, entity)
-        cache.referrers.remove(entity.id)
+        // TODO: [referrerEntities] could be removed if there were an easy way to list references
+        cache.referrerEntities.remove(entity.id)?.forEach {
+            cache.referrers[it]?.remove(entity.id)
+        }
```

Comment:
This is the only place where `referrerEntities` is used.  You could just as easily use something akin to:
```
entity.references.filter { it is Referrers }.forEach { cache.referrers[it.reference].remove(entity.id) }
```
and remove `referrerEntities`.  I could not figure out a clean way to get `entity.references`, but that would be super helpful in general.